mainwindow.h

```diff
@@ -36,8 +36,8 @@ private slots:
     void on_localModels_activated(int index);
 
 private:
-    Ui::MainWindow *ui;
-    MarianInterface * translator_; // Initialised to nullptr. Can't use smart ptrs due to destructor needing to run
+    std::unique_ptr<Ui::MainWindow> ui_;
```

Comment:
Could we use `QPointer` for QObjects?

Reply:
QPointer doesn't delete, it just lets you know the object was deleted.  

https://stackoverflow.com/questions/40346393/should-i-use-qscopedpointer-or-stdunique-ptr

Reply:
Ok, then I should refactor the whole code to not use the (useless) QPointer, but to instead use unique_ptr